reimbursable_with_tax/tests/test_reimbursable.py

```diff
@@ -0,0 +1,99 @@
+        self.currency_usd = self.env.ref('base.USD')
+        self.main_company = self.env.ref('base.main_company')
+        self.env.cr.execute(
+            """UPDATE res_company SET currency_id = %s
+            WHERE id = %s""",
+            (self.main_company.id, self.currency_eur.id),
+        )
+        self.journal = self.env['account.journal'].create({
+            'name': 'Journal',
+            'type': 'purchase',
+            'code': 'TEST',
+        })
+        self.partner = self.env['res.partner'].create({
+            'supplier': True,
+            'name': 'Partner',
+        })
+        self.reimbursable_partner = self.env['res.partner'].create({
+            'supplier': True,
+            'name': 'Reimbursable Partner',
+        })
+        self.product = self.env['product.product'].create({
+            'name': 'Product',
+        })
+        self.reimbursable = self.env['product.product'].create({
+            'name': 'Reimbursable',
+            'type': 'service',
+        })
+        self.reimbursable_tax = self.env['account.tax'].search([], limit=1)
+
+    def create_invoice_vals(self):
+        return {
+            'partner_id': self.partner.id,
+            'account_id': self.partner.property_account_payable_id.id,
+            'journal_id': self.journal.id,
+            'invoice_line_ids': [(0, 0, {
+                'product_id': self.product.id,
+                'account_id':
+                    self.product.product_tmpl_id.get_product_accounts(
+
+                    )['expense'].id,
+                'name': self.product.name,
+                'price_unit': 100,
+            })]
+        }
+
+    def test_create_reimbursable_with_tax(self):
+        invoice = self.complete_check_invoice(100)
+        self.assertFalse(invoice.tax_line_ids)
+        invoice._onchange_invoice_line_reimbursable_ids()
+        self.assertTrue(invoice.tax_line_ids)
+        with self.assertRaises(UserError):
+            invoice.action_invoice_open()
+        for tax in invoice.tax_line_ids:
+            self.assertEqual(1000, tax.sequence)
+            tax.write({
+                'tax_invoice_manual': 'invoice number',
+                'tax_date_manual': fields.Date.today(),
+            })
+        invoice.action_invoice_open()
+
+    def complete_check_invoice(self, amount):
+        invoice = self.env['account.invoice'].with_context(
+            default_type='in_invoice', type='in_invoice',
+            journal_type='purchase'
+        ).create(self.create_invoice_vals())
+        self.assertEqual(invoice.amount_total, 100)
+        self.assertEqual(invoice.reimbursable_count, 0)
+        self.assertEqual(invoice.executable_total, 100)
+        reimbursable = self.env['account.invoice.reimbursable'].new({
+            'invoice_id': invoice.id,
+            'amount': amount
+        })
+        reimbursable.partner_id = self.reimbursable_partner
+        self.assertFalse(reimbursable.account_id)
```

Comment:
In the test, should you test about base amount also? I think it is what we are after.